plugin/pkg/shoot/oidc/openidconnectpreset/admission_test.go

```diff
@@ -162,13 +171,18 @@ var _ = Describe("OpenID Connect Preset", func() {
 
 		})
 
-		Context("should mutate the result", func() {
+		Context("should mutate the result for shoot kubernetes <= 1.32", func() {
 			var (
-				expected *core.Shoot
+				expected     *core.Shoot
+				shootLess132 *core.Shoot // shoot version <1.32
 			)
 
 			BeforeEach(func() {
-				expected = shoot.DeepCopy()
+				shootNameLess132 := "shoot1"
+				shootLess132 = shoot.DeepCopy()
+				shootLess132.Name = shootNameLess132
+				shootLess132.Spec.Kubernetes.Version = "1.31.0"
+				expected = shootLess132.DeepCopy()
```

Comment:
I think you could just reuse the shoot variable here. It will get reset back to its initial value for each `It` since it is assigned in the first `BeforeEach`
```suggestion
				shoot.Spec.Kubernetes.Version = "1.31.0"
				expected = shoot.DeepCopy()
```

Reply:
hmm not sure about this in case tests would run in parallel

Reply:
Yeah, I had the same concern, but I think ginkgo actually works around that by running each parallel test in separate processes: https://onsi.github.io/ginkgo/#mental-model-how-ginkgo-runs-parallel-specs

> As i can see in each contexts BeforeEach we are doing deep copy probably because of this
If you mean the `expected` variable - the shoot is deep copied into it because the invocation of the Admit function could mutate the shoot variable. So we want to keep the expected value separate.

Reply:
ah ok yeah, i read this quite a long in past, forgot about it, thx